sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clientTelemetry/ClientTelemetry.java

```diff
@@ -0,0 +1,230 @@
+    private volatile boolean isClosed;
+    private volatile boolean isClientTelemetryEnabled;
+    private static String AZURE_VM_METADATA = "http://169.254.169.254:80/metadata/instance?api-version=2020-06-01";
+
+    private static final double PERCENTILE_50 = 50.0;
+    private static final double PERCENTILE_90 = 90.0;
+    private static final double PERCENTILE_95 = 95.0;
+    private static final double PERCENTILE_99 = 99.0;
+    private static final double PERCENTILE_999 = 99.9;
+    private final int clientTelemetrySchedulingSec;
+
+    public ClientTelemetry(Boolean acceleratedNetworking,
+                           String clientId,
+                           String processId,
+                           String userAgent,
+                           ConnectionMode connectionMode,
+                           String globalDatabaseAccountName,
+                           String applicationRegion,
+                           String hostEnvInfo,
+                           HttpClient httpClient,
+                           boolean isClientTelemetryEnabled
+    ) {
+        clientTelemetryInfo = new ClientTelemetryInfo(clientId, processId, userAgent, connectionMode,
+            globalDatabaseAccountName, applicationRegion, hostEnvInfo, acceleratedNetworking);
+        this.isClosed = false;
+        this.httpClient = httpClient;
+        this.isClientTelemetryEnabled = isClientTelemetryEnabled;
+        this.clientTelemetrySchedulingSec = Configs.getClientTelemetrySchedulingInSec();
+    }
+
+    public ClientTelemetryInfo getClientTelemetryInfo() {
+        return clientTelemetryInfo;
+    }
+
+    public static void recordValue(DoubleHistogram doubleHistogram, long value) {
+        try {
+            doubleHistogram.recordValue(value);
+        } catch (Exception ex) {
+            logger.warn("Error while recording value for client telemetry", ex.getMessage());
+        }
+    }
+
+    public static void recordValue(DoubleHistogram doubleHistogram, double value) {
+        try {
+            doubleHistogram.recordValue(value);
+        } catch (Exception ex) {
+            logger.warn("Error while recording value for client telemetry", ex.getMessage());
+        }
+    }
+
+    public void init() {
+        loadAzureVmMetaData();
+        sendClientTelemetry().subscribe();
+    }
+
+    public void close() {
+        this.isClosed = true;
+        this.executor.shutdown();
+        logger.debug("GlobalEndpointManager closed.");
+    }
+
+    private Mono<Void> sendClientTelemetry() {
+        return Mono.delay(Duration.ofSeconds(clientTelemetrySchedulingSec))
+            .flatMap(t -> {
+                if (this.isClosed) {
+                    logger.warn("client already closed");
+                    return Mono.empty();
+                }
+
+                if(!Configs.isClientTelemetryEnabled(this.isClientTelemetryEnabled)) {
+                    logger.trace("client telemetry not enabled");
```

Comment:
This is done intentionally, till the time we enable telemetry by default i don't want cx to get this log. Otherwise every customer will be getting this every 10 min , and might create a cri with question.